src/uct/ib/base/ib_iface.c

```diff
@@ -385,25 +403,27 @@ void uct_ib_iface_address_pack(uct_ib_iface_t *iface, uct_ib_address_t *ib_addr)
 {
     uct_ib_address_pack_params_t params;
 
-    params.flags     = uct_ib_iface_address_pack_flags(iface);
-    params.gid       = &iface->gid_info.gid;
-    params.lid       = uct_ib_iface_port_attr(iface)->lid;
-    params.roce_info = &iface->gid_info.roce_info;
+    params.flags      = uct_ib_iface_address_pack_flags(iface);
+    params.gid        = &iface->gid_info.gid;
+    params.lid        = uct_ib_iface_port_attr(iface)->lid;
+    params.roce_info  = &iface->gid_info.roce_info;
     /* to suppress gcc 4.3.4 warning */
-    params.path_mtu  = UCT_IB_ADDRESS_INVALID_PATH_MTU;
-    params.gid_index = UCT_IB_ADDRESS_INVALID_GID_INDEX;
+    params.path_mtu   = UCT_IB_ADDRESS_INVALID_PATH_MTU;
+    params.gid_index  = UCT_IB_ADDRESS_INVALID_GID_INDEX;
+    params.pkey_value = iface->pkey_value;
     uct_ib_address_pack(&params, ib_addr);
 }
 
 void uct_ib_address_unpack(const uct_ib_address_t *ib_addr, uint16_t *lid,
                            union ibv_gid *gid, uint8_t *gid_index,
-                           enum ibv_mtu *path_mtu)
+                           enum ibv_mtu *path_mtu, uint16_t *pkey_value)
```

Comment:
maybe this function could return *uct_ib_address_pack_params_t ? @evgeny-leksikov  WDYT?
there are already so many values it has to return?
(can be in different PR if it's big change)

Reply:
👍 but then need to rename `uct_ib_address_pack_params_t` to smth common and probably unite `UCT_IB_ADDRESS_xx` and `UCT_IB_ADDRESS_PACK_xx` flags. I did not do this change because it'a a big and should be bedefinitely done separately.